src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,98 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_SIZE = 10;
+    private static final double MULTIPLIER = 1.5;
+    private int size;
+    private T[] array;
+
+    public ArrayList() {
+        array = (T[]) new Object[DEFAULT_SIZE];
+    }
+
     @Override
     public void add(T value) {
-
+        growIfSizeFull();
+        array[size++] = value;
     }
 
     @Override
     public void add(T value, int index) {
-
+        indexVerifyForAdd(index);
+        growIfSizeFull();
+        System.arraycopy(array, index, array, index + 1, size - index);
+        array[index] = value;
+        size++;
```

Comment:
Regarding Oleh's comment: After such change you should be able to remove indexVerifyForAdd method
```suggestion
        if (index == size) {
            add(value);
            return;
        }
        indexVerify(index);
        growIfSizeFull();
        System.arraycopy(array, index, array, index + 1, size - index);
        array[index] = value;
        size++;
```

Reply:
thanks, now I understand🙂